go/private/tools/path.bzl

```
@@ -75,15 +81,18 @@ def _go_path_impl(ctx):
         # src_dir is the path to the directory holding the source.
         # Paths to embedded sources will be relative to this path.
         src_dir = None
+        if len(pkg.root_dir) > 0:
+            src_dir = pkg.root_dir[0]
 
         for f in pkg.srcs:
-            src_dir = f.dirname
             dst = pkg.dir + "/" + f.basename
             _add_manifest_entry(manifest_entries, manifest_entry_map, inputs, f, dst)
         for f in pkg.embedsrcs:
             if src_dir == None:
                 fail("cannot relativize {}: src_dir is unset".format(f.path))
             embedpath = paths.relativize(f.path, f.root.path)
+            if embedpath.startswith(ctx.bin_dir.path + "/"):
+                embedpath = embedpath[len(ctx.bin_dir.path) + 1:]
             dst = pkg.dir + "/" + paths.relativize(embedpath, src_dir)
```

```suggestion
            dst = pkg.dir + "/" + paths.relativize(embedpath.lstrip(ctx.bin_dir.path + "/"), src_dir.lstrip(ctx.bin_dir.path + "/"))
```
I make this change locally without all other changes you made, and run the tests in this PR, they all passed. Maybe this is all we need?

---

@linzhp Thank you so much for looking into this,  you're right, and that's a better fix. I already applied the changes